aws/retry/middleware_test.go

```diff
@@ -193,6 +198,20 @@ func TestAttemptMiddleware(t *testing.T) {
 					MaxAttempts: 3,
 				},
 			},
+			ExpectMetadata: requestAttempts{Attempts: []requestAttempt{
+				{
+					Retried: true,
+				},
+				{
+					Error:     mockRetryableError{b: true},
+					Retryable: true,
+					Retried:   true,
+				},
+				{
+					Error:     mockRetryableError{b: true},
+					Retryable: true,
+				},
+			}},
```

Comment:
Are these in reverse order? I'd expect the order to be [first, second, thrid] but this looks to be [thrid, second, first].  This might be an artifact of the `defer` within the for loop in middleware.go

Reply:
yeah they are in reverse order. Thought the last request is most important in most cases, and thus used defer.